### ISSUE TITLE
EWS some app sharing type message can not receive

### ISSUE THREAD

**Issue by xufan6** [blueset/ehForwarderBot#25]
```
2017-02-11 18:49:52,749: plugins.eh_wechat_slave.wechat_msg_meta [DEBUG]
    Raw message: {'Type': 'Sharing', 'RecommendInfo': {'QQNum': 0, 'Sex': 0, 'Ticket': '', 'NickName': '', 'Alias': '', 'OpCode': 0, 'AttrStatus': 0, 'City': '', 'Signature': '', 'VerifyFlag': 0, 'UserName': '', 'Scene': 0, 'Content': '', 'Province': ''}, 'FileName': '缸鸭狗(湖滨店)', 'ImgStatus': 2, 'ForwardFlag': 0, 'SubMsgType': 0, 'NewMsgId': 6549964xxxxxxx, 'FileSize': '', 'AppMsgType': 5, 'MsgType': 49, 'AppInfo': {'Type': 0, 'AppID': 'wx8e251222d6836a60'}, 'FromUserName': '@4d953d2be6d3xxxxx', 'ToUserName': '@099639dbd1xxxxxx', 'StatusNotifyCode': 0, 'Content': '<?xml version="1.0"?>\n<msg>\n\t<appmsg appid="wx8e251222d6836a60" sdkver="0">\n\t\t<title>缸鸭狗(湖滨店)</title>\n\t\t<des>★★★★\n￥54/人\n湖滨 小吃快餐\n延安路255号龙翔里二弄104室</des>\n\t\t<username />\n\t\t<action>view</action>\n\t\t<type>5</type>\n\t\t<showtype>0</showtype>\n\t\t<content />\n\t\t<url>http://m.dianping.com/appshare/shop/21640420</url>\n\t\t<lowurl />\n\t\t<dataurl />\n\t\t<lowdataurl />\n\t\t<contentattr>0</contentattr>\n\t\t<streamvideo>\n\t\t\t<streamvideourl />\n\t\t\t<streamvideototaltime>0</streamvideototaltime>\n\t\t\t<streamvideotitle />\n\t\t\t<streamvideowording />\n\t\t\t<streamvideoweburl />\n\t\t\t<streamvideothumburl />\n\t\t\t<streamvideoaduxinfo />\n\t\t\t<streamvideopublishid />\n\t\t</streamvideo>\n\t\t<canvasPageItem>\n\t\t\t<canvasPageXml><![CDATA[]]></canvasPageXml>\n\t\t</canvasPageItem>\n\t\t<appattach>\n\t\t\t<attachid />\n\t\t\t<cdnthumburl>305a0201000453305102010002048356c25402033d11ff020450c9c0b70204589dbda6042f6175706170706d73675f613336373164613763306364653435305f313438363733323731303730325f3131383731380201000201000400</cdnthumburl>\n\t\t\t<cdnthumbmd5>a83469379d593f39adb6af79eb0e299d</cdnthumbmd5>\n\t\t\t<cdnthumblength>15483</cdnthumblength>\n\t\t\t<cdnthumbheight>110</cdnthumbheight>\n\t\t\t<cdnthumbwidth>110</cdnthumbwidth>\n\t\t\t<cdnthumbaeskey>30fd124e6182401f92571ccfac103767</cdnthumbaeskey>\n\t\t\t<aeskey>30fd124e6182401f92571ccfac103767</aeskey>\n\t\t\t<encryver>1</encryver>\n\t\t\t<fileext />\n\t\t\t<islargefilemsg>0</islargefilemsg>\n\t\t</appattach>\n\t\t<extinfo />\n\t\t<androidsource>2</androidsource>\n\t\t<thumburl />\n\t\t<mediatagname />\n\t\t<messageaction><![CDATA[]]></messageaction>\n\t\t<messageext><![CDATA[]]></messageext>\n\t\t<emoticongift>\n\t\t\t<packageflag>0</packageflag>\n\t\t\t<packageid />\n\t\t</emoticongift>\n\t\t<emoticonshared>\n\t\t\t<packageflag>0</packageflag>\n\t\t\t<packageid />\n\t\t</emoticonshared>\n\t\t<weappinfo>\n\t\t\t<pagepath><![CDATA[null]]></pagepath>\n\t\t\t<username />\n\t\t\t<appid />\n\t\t</weappinfo>\n\t\t<designershared>\n\t\t\t<designeruin>0</designeruin>\n\t\t\t<designername>null</designername>\n\t\t\t<designerrediretcturl>null</designerrediretcturl>\n\t\t</designershared>\n\t\t<emotionpageshared>\n\t\t\t<tid>0</tid>\n\t\t\t<title>null</title>\n\t\t\t<desc>null</desc>\n\t\t\t<iconUrl>null</iconUrl>\n\t\t\t<secondUrl>null</secondUrl>\n\t\t\t<pageType>0</pageType>\n\t\t</emotionpageshared>\n\t\t<webviewshared>\n\t\t\t<shareUrlOriginal />\n\t\t\t<shareUrlOpen />\n\t\t\t<jsAppId />\n\t\t\t<publisherId />\n\t\t</webviewshared>\n\t\t<template_id>0</template_id>\n\t\t<statextstr>GhQKEnd4OGUyNTEyMjJkNjgzNmE2MA==</statextstr>\n\t\t<md5 />\n\t\t<md5>a83469379d593f39adb6af79eb0e299d</md5>\n\t</appmsg>\n\t<fromusername>cy509529</fromusername>\n\t<scene>0</scene>\n\t<appinfo>\n\t\t<version>70</version>\n\t\t<appname>大众点评</appname>\n\t</appinfo>\n\t<commenturl></commenturl>\n</msg>\n', 'CreateTime': 1486810190, 'VoiceLength': 0, 'Url': 'http://m.dianping.com/appshare/shop/21640420', 'MsgId': '65499643xxxx', 'OriContent': '', 'Ticket': '', 'Status': 3, 'Text': '缸鸭狗(湖滨店)', 'MediaId': '', 'ImgHeight': 0, 'ImgWidth': 0, 'StatusNotifyUserName': '', 'HasProductId': 0, 'PlayLength': 0}
```
```
2017-02-11 18:59:23,217: plugins.eh_wechat_slave.wechat_msg_meta [DEBUG]
    Raw message: {'FileSize': '', 'StatusNotifyUserName': '', 'MsgId': '33030566xxxx', 'MediaId': '', 'OriContent': '', 'StatusNotifyCode': 0, 'FromUserName': '@099639dbd1xxxxxx', 'ImgWidth': 0, 'Url': 'http://m.dianping.com/appshare/shop/21640420', 'VoiceLength': 0, 'PlayLength': 0, 'AppInfo': {'Type': 0, 'AppID': 'wx8e251222d6836a60'}, 'FileName': '缸鸭狗(湖滨店)', 'ToUserName': 'filehelper', 'HasProductId': 0, 'AppMsgType': 5, 'Type': 'Sharing', 'SubMsgType': 0, 'CreateTime': 1486810762, 'ImgStatus': 2, 'Content': '<?xml version="1.0"?>\n<msg>\n\t<appmsg appid="wx8e251222d6836a60" sdkver="0">\n\t\t<title>缸鸭狗(湖滨店)</title>\n\t\t<des>★★★★\n￥54/人\n湖滨 小吃快餐\n延安路255号龙翔里二弄104室</des>\n\t\t<username />\n\t\t<action>view</action>\n\t\t<type>5</type>\n\t\t<showtype>0</showtype>\n\t\t<content />\n\t\t<url>http://m.dianping.com/appshare/shop/21640420</url>\n\t\t<lowurl />\n\t\t<dataurl />\n\t\t<lowdataurl />\n\t\t<contentattr>0</contentattr>\n\t\t<streamvideo>\n\t\t\t<streamvideourl />\n\t\t\t<streamvideototaltime>0</streamvideototaltime>\n\t\t\t<streamvideotitle />\n\t\t\t<streamvideowording />\n\t\t\t<streamvideoweburl />\n\t\t\t<streamvideothumburl />\n\t\t\t<streamvideoaduxinfo />\n\t\t\t<streamvideopublishid />\n\t\t</streamvideo>\n\t\t<canvasPageItem>\n\t\t\t<canvasPageXml><![CDATA[]]></canvasPageXml>\n\t\t</canvasPageItem>\n\t\t<appattach>\n\t\t\t<attachid />\n\t\t\t<cdnthumburl>305a0201000453305102010002048356c25402033d11ff020450c9c0b70204589dc1e2042f6175706170706d73675f393237623035356462656438303665665f313438363733333739343332325f3131383736380201000201000400</cdnthumburl>\n\t\t\t<cdnthumbmd5>74f7ab7f59a22b5be6115100b2a97deb</cdnthumbmd5>\n\t\t\t<cdnthumblength>28648</cdnthumblength>\n\t\t\t<cdnthumbheight>120</cdnthumbheight>\n\t\t\t<cdnthumbwidth>120</cdnthumbwidth>\n\t\t\t<cdnthumbaeskey>13dff4f17ab448599c76d21c5c6330e2</cdnthumbaeskey>\n\t\t\t<aeskey>13dff4f17ab448599c76d21c5c6330e2</aeskey>\n\t\t\t<encryver>1</encryver>\n\t\t\t<fileext />\n\t\t\t<islargefilemsg>0</islargefilemsg>\n\t\t</appattach>\n\t\t<extinfo />\n\t\t<androidsource>3</androidsource>\n\t\t<thumburl />\n\t\t<mediatagname />\n\t\t<messageaction><![CDATA[]]></messageaction>\n\t\t<messageext><![CDATA[]]></messageext>\n\t\t<emoticongift>\n\t\t\t<packageflag>0</packageflag>\n\t\t\t<packageid />\n\t\t</emoticongift>\n\t\t<emoticonshared>\n\t\t\t<packageflag>0</packageflag>\n\t\t\t<packageid />\n\t\t</emoticonshared>\n\t\t<weappinfo>\n\t\t\t<pagepath><![CDATA[null]]></pagepath>\n\t\t\t<username />\n\t\t\t<appid />\n\t\t</weappinfo>\n\t\t<designershared>\n\t\t\t<designeruin>0</designeruin>\n\t\t\t<designername>null</designername>\n\t\t\t<designerrediretcturl>null</designerrediretcturl>\n\t\t</designershared>\n\t\t<emotionpageshared>\n\t\t\t<tid>0</tid>\n\t\t\t<title>null</title>\n\t\t\t<desc>null</desc>\n\t\t\t<iconUrl>null</iconUrl>\n\t\t\t<secondUrl>null</secondUrl>\n\t\t\t<pageType>0</pageType>\n\t\t</emotionpageshared>\n\t\t<webviewshared>\n\t\t\t<shareUrlOriginal />\n\t\t\t<shareUrlOpen />\n\t\t\t<jsAppId />\n\t\t\t<publisherId />\n\t\t</webviewshared>\n\t\t<template_id>0</template_id>\n\t\t<statextstr>GhQKEnd4OGUyNTEyMjJkNjgzNmE2MA==</statextstr>\n\t\t<md5 />\n\t\t<md5>74f7ab7f59a22b5be6115100b2a97deb</md5>\n\t</appmsg>\n\t<fromusername>xufan6</fromusername>\n\t<scene>0</scene>\n\t<appinfo>\n\t\t<version>70</version>\n\t\t<appname>大众点评</appname>\n\t</appinfo>\n\t<commenturl></commenturl>\n</msg>\n', 'ForwardFlag': 0, 'RecommendInfo': {'UserName': '', 'City': '', 'Alias': '', 'VerifyFlag': 0, 'Content': '', 'Province': '', 'Sex': 0, 'Scene': 0, 'Signature': '', 'NickName': '', 'AttrStatus': 0, 'Ticket': '', 'QQNum': 0, 'OpCode': 0}, 'NewMsgId': 33030566xxxxxxx, 'Text': '缸鸭狗(湖滨店)', 'Status': 3, 'ImgHeight': 0, 'Ticket': '', 'MsgType': 49}
```

**Comment by blueset** [comment]
How is such message looks like on WeChat mobile client?

**Comment by xufan6** [comment]
![image](https://cloud.githubusercontent.com/assets/364394/22854969/dbc3d66c-f0b2-11e6-8581-4f532687d174.png)

**Comment by blueset** [comment]
Fixed in 1.4.4-dev.12 (4b0a7f8).
